aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -752,33 +751,38 @@ Tensor select_sparse_csr(const Tensor& self, int64_t dim, int64_t index) {
 
   // Selecting batch dimension
   if (dim < self.dim() - 2) {
-    if (self.layout() == kSparseBsr) {
-      return at::native::_sparse_bsr_tensor_unsafe(
-          self.crow_indices().select(dim, index),
-          self.col_indices().select(dim, index),
-          self.values().select(dim, index),
-          new_sizes,
-          optTypeMetaToScalarType(options.dtype_opt()),
-          options.layout_opt(),
-          options.device_opt(),
-          options.pinned_memory_opt());
-    }
-    return at::native::_sparse_csr_tensor_unsafe(
-        self.crow_indices().select(dim, index),
-        self.col_indices().select(dim, index),
-        self.values().select(dim, index),
-        new_sizes,
-        optTypeMetaToScalarType(options.dtype_opt()),
-        options.layout_opt(),
-        options.device_opt(),
-        options.pinned_memory_opt());
+    return AT_DISPATCH_ROW_SPARSE_COMPRESSED_LAYOUTS(
+        self.layout(),
+        "select",
+        [&]() {
+          return at::native::_sparse_compressed_tensor_unsafe(
+              self.crow_indices().select(dim, index),
+              self.col_indices().select(dim, index),
```

Comment:
nit: we could extract compressed and plain indices in the dispatch part and then call the constructor. Imho, might be more transparent and with less code.